src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter((person) =>
+    century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm'
+  );
+
+  const sumAges = men.reduce(
+    (sum, person) => sum + (person.died - person.born),
+    0
+  );
+
+  return Math.round((sumAges / men.length) * 100) / 100;
```

Comment:
Possible variable as well?

Reply:
Hi Amit, thank you for the suggestions. I tried to implement them. Let me know if all is good. Cheers!